backend/mcp_backend/routers/mcp.py

```
@@ -0,0 +1,963 @@
+                status_code=500, detail=f"Failed to restart server {server_name}"
+            )
+
+        return {"message": f"Built-in server {server_name} restarted successfully"}
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        log.exception(f"Error restarting built-in MCP server: {e}")
+        raise HTTPException(
+            status_code=500, detail=f"Error restarting built-in server: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)